crates/fuel-core/src/service/adapters/gas_price_adapters.rs

```diff
@@ -39,6 +58,44 @@ impl L2Data for OnChainIterableKeyValueView {
     }
 }
 
+impl GasPriceData for Database<GasPriceDatabase> {
+    fn latest_height(&self) -> Option<BlockHeight> {
+        HistoricalView::latest_height(self)
+    }
+}
+
+impl MetadataStorage for Database<GasPriceDatabase> {
```

Comment:
Why do we want to duplicate the implementation here instead of deriving the blanket implementation from gas price service crate?

Reply:
addressed in 417459d415